test/either.test.js

```diff
@@ -0,0 +1,219 @@
+	);
+});
+
+qunit.test("#map", (assert) => {
+	assert.equal(
+		either(1).map(identity)._inspect(),
+		either(1)._inspect(),
+		"should follow the identity law given a Either Right monad"
+	);
+
+	assert.equal(
+		either(1).map(x => twice(inc(x)))._inspect(),
+		either(1).map(inc).map(twice)._inspect(),
+		"should follow the composition law given a Either Right monad"
+	);
+
+	const operation = sinon.fake();
+
+	assert.equal(
+		either.Left(1).map(operation)._inspect(),
+		either.Left(1)._inspect(),
+		"should return a Either Left monad given a Either Left monad"
+	);
+
+
+	assert.equal(
+		operation.called,
+		false,
+		"should perform no operation given a Either Left monad"
+	);
+});
+
+qunit.test("#chain", (assert) => {
+	assert.equal(
+		either({ name: "john" }).chain(eitherProp("name"))._inspect(),
+		either("john")._inspect(),
+		"should return an Either Right monad with 'john' as value"
+	);
+
+	const operation = sinon.fake();
+
+	assert.equal(
+		either.Left({ name: "john" }).chain(operation)._inspect(),
+		either.Left({ name: "john" })._inspect(),
+		"should return an Either Left monad given an Either Left monad"
+	);
+
+	assert.equal(
+		either({ name: "john" }).flatMap(eitherProp("name"))._inspect(),
+		either("john")._inspect(),
+		"should return an Either Right monad with 'john' as value"
+	);
+
+	assert.equal(
+		either.Left({ name: "john" }).flatMap(operation)._inspect(),
+		either.Left({ name: "john" })._inspect(),
+		"should return an Either Left monad given an Either Left monad"
+	);
+
+	assert.equal(
+		either({ name: "john" }).bind(eitherProp('name'))._inspect(),
+		either("john")._inspect(),
+		"should return an Either Right monad with 'john' as value"
+	);
+
+	assert.equal(
+		either.Left({ name: "john" }).bind(operation)._inspect(),
+		either.Left({ name: "john" })._inspect(),
+		"should return an Either Left monad given an Either Left monad"
+	);
+
+	assert.equal(
+		operation.called,
+		false,
+		"should perform no operation given a Either Left monad"
+	);
+});
+
+qunit.test("#ap", (assert) => {
+	assert.equal(
+		either(inc).ap(either(2))._inspect(),
+		either(3)._inspect(),
+		"should return a Either Right 4 monad given an Either Right inc monad and an Either Right 2 monad"
+	);
+
+	const operation = sinon.fake();
+
+	assert.equal(
+		either(operation).ap(either.Left(2))._inspect(),
+		either.Left(2)._inspect(),
+		"should return a Either Left 2 monad given an Either Right op monad and an Either Left 2 monad"
+	);
+
+	assert.equal(
+		either.Left(operation).ap(either(2))._inspect(),
+		either.Left(operation)._inspect(),
+		"should return a Either Left op monad given an Either Left op monad and an Either Left 2 monad"
+	);
+
+	assert.equal(
+		operation.called,
+		false,
+		"should perform no operation given a Either Left monad"
+	);
+});
+
+qunit.test("#concat", (assert) => {
+	assert.deepEqual(
+		either([1, 2]).concat([3]),
+		[[1, 2, 3]],
+		"should return an array given a Either Left monad and [3]"
```

Comment:
I believe this is an Either:Right monad?

Reply:
Yeah, my bad